### PR TITLE
[editor] fix(pluginToolbarAlignment): calculated max and min left of toolbar

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/common/web/src/Base/createBaseToolbar.jsx [hunk 1]
```diff
@@ -201,21 +203,37 @@ export default function createToolbar({
...
+        '--offset-top': `${top}px`,
+        '--offset-left': `${left}px`,
         transform: 'scale(1)',
       };
     }
 
+    calculateLeftOffset = (left, maxLeft) => {
+      if (this.context?.languageDir === 'ltr') {
+        if (left < 0) {
+          return -toolbarLeftOffset * 2;
```
Comment (lxgreen) [inline packages/common/web/src/Base/createBaseToolbar.jsx]: `toolbarLeftOffset` is defined as local var in caller function; available via closure, not guaranteed to be available if the `calculateLeftOffset` called from another place.

File: packages/common/web/src/Base/createBaseToolbar.jsx [hunk 2]
```diff
@@ -201,21 +203,37 @@ export default function createToolbar({
...
-          offsetParentLeft -
-          toolbarWidth / 2 +
-          x}px`,
+        '--offset-top': `${top}px`,
+        '--offset-left': `${left}px`,
         transform: 'scale(1)',
       };
     }
 
+    calculateLeftOffset = (left, maxLeft) => {
```
Comment (lxgreen) [inline packages/common/web/src/Base/createBaseToolbar.jsx]:
```suggestion
    calculateLeftOffset = (left, maxLeft) => {
      const isLtr = this.context?.languageDir === 'ltr';
      const outOfMargins =isLtr ? (left < 0) : (left > maxLeft);
      if (outOfMargins) {
        return -toolbarLeftOffset * 2;
      }
      if (isLtr) {
        return Math.min(left, maxLeft);
      }
      return left < 0 ? maxLeft : left;
    }
```

File: packages/common/web/src/Base/createBaseToolbar.jsx [hunk 3]
```diff
@@ -201,21 +203,37 @@ export default function createToolbar({
...
+      if (this.context?.languageDir === 'ltr') {
+        if (left < 0) {
+          return -toolbarLeftOffset * 2;
+        } else {
+          return Math.min(left, maxLeft);
+        }
+      } else {
+        // eslint-disable-next-line no-lonely-if
+        if (left > maxLeft) {
```
Comment (lxgreen) [inline packages/common/web/src/Base/createBaseToolbar.jsx]: all the `if-else` nesting is still not readable; please see my suggestion and see the differences

File: packages/common/web/src/Base/createBaseToolbar.jsx [hunk 4]
```diff
@@ -16,7 +17,8 @@ import Panel from '../Components/Panel';
 import toolbarStyles from '../../statics/styles/plugin-toolbar.scss';
 import buttonStyles from '../../statics/styles/plugin-toolbar-button.scss';
 
-const toolbarOffset = 12;
+const toolbarTopOffset = 12;
```
Comment (lxgreen) [inline packages/common/web/src/Base/createBaseToolbar.jsx]: suggest to move these consts to `consts` module

File: packages/common/web/src/Base/createBaseToolbar.jsx (hunk 3 above)
Comment (maimonav) [reply packages/common/web/src/Base/createBaseToolbar.jsx]: Done, thanks.

File: packages/common/web/src/Base/createBaseToolbar.jsx [hunk 5]
```diff
@@ -201,17 +201,14 @@ export default function createToolbar({
       const offsetParentTop = offsetParentRect.top;
       const offsetParentLeft = offsetParentRect.left;
       const boundingRect = pubsub.get('boundingRect');
+      const top = boundingRect.top - toolbarHeight - toolbarOffset - offsetParentTop + y;
+      let left =
+        boundingRect.left + boundingRect.width / 2 - offsetParentLeft - toolbarWidth / 2 + x;
+      const maxLeft = boundingRect.right - toolbarWidth;
+      left = left >= 0 && left <= maxLeft ? left : left < 0 ? 0 : maxLeft;
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/common/web/src/Base/createBaseToolbar.jsx]: Done, thanks.

File: packages/common/web/src/Base/createBaseToolbar.jsx (hunk 1 above)
Comment (maimonav) [reply packages/common/web/src/Base/createBaseToolbar.jsx]: I moved to toolbarLeftOffset and toolbarTopOffset to consts. Thanks.

File: packages/common/web/src/Base/createBaseToolbar.jsx (hunk 2 above)
Comment (maimonav) [reply packages/common/web/src/Base/createBaseToolbar.jsx]: Thanks for the explained refactor.

File: packages/common/web/src/Base/createBaseToolbar.jsx (hunk 4 above)
Comment (maimonav) [reply packages/common/web/src/Base/createBaseToolbar.jsx]: Done, thanks.

**Pull request merged**